openedx/features/content_tagging/tasks.py

```diff
@@ -0,0 +1,148 @@
+"""
+Defines asynchronous celery task for auto-tagging content
+"""
+
+import logging
+
+from celery import shared_task
+from celery_utils.logged_task import LoggedTask
+from django.contrib.auth import get_user_model
+from edx_django_utils.monitoring import set_code_owner_attribute
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from openedx_tagging.core.tagging.models import Taxonomy
+
+from xmodule.modulestore.django import modulestore
+
+from . import api
+
+LANGUAGE_TAXONOMY_ID = -1
+
+log = logging.getLogger(__name__)
+User = get_user_model()
+
+
+def _has_taxonomy(taxonomy: Taxonomy, content_object) -> bool:
+    """
+    Return True if this Taxonomy have some Tag set in the content_object
+    """
+    _exausted = object()
+
+    content_tags = api.get_content_tags(object_id=content_object, taxonomy_id=taxonomy.id)
+    return next(content_tags, _exausted) is not _exausted
+
+
+def _update_tags(content_object, lang) -> None:
+    """
+    Update the tags for a content_object.
+
+    If the content_object already have a tag for the language taxonomy, it will be skipped.
```

Comment:
It is to avoid auto-tagging (overriding) content manually tagged or changed by the user. This is stated in the issue description [here](https://github.com/openedx/modular-learning/issues/79).

The way to avoid this is to create a new flag in the ObjectTag model to let us know if it was created by the system or edited by the user.

Anyway, I will update the function name as you suggested! Thank you!

Reply:
Done ce23eb19aeb442e8878e070484eb0a82a887f31d